gocov/convert/convert.go

```diff
@@ -37,18 +39,33 @@ import (
 	"golang.org/x/tools/cover"
 )
 
+const (
+	goModFilename = "go.mod"
+)
+
 func marshalJson(w io.Writer, packages []*gocov.Package) error {
 	return json.NewEncoder(w).Encode(struct{ Packages []*gocov.Package }{packages})
 }
 
-type packagesCache map[string]*build.Package
+type PackageInfo struct {
+	Dir        string
+	ImportPath string
+}
+
+type packagesCache map[string]PackageInfo
 
 func ConvertProfiles(filenames ...string) ([]byte, error) {
 	var (
 		ps       gocovutil.Packages
 		packages = make(packagesCache)
 	)
 
+	goModContent, err := os.ReadFile(goModFilename)
+	if err != nil {
+		return nil, fmt.Errorf("getting module name: read go.mod: %w", err)
+	}
```

Comment:
I think this is a breaking change. `gocov convert` will fail if it's not run in the top level of the module. I don't think that's currently the case is it?

Instead of reading the `go.mod` file like this, could we use [golang.org/x/tools/go/packages](https://pkg.go.dev/golang.org/x/tools/go/packages#Load) to load the packages? I'm thinking something like this:
 - Parse the profiles using golang.org/x/tools/cover
 - Iterate through each profile, and collect the package names
 - Call golang.org/x/tools/go/packages.Load with the package names
 - Iterate through each profile again, converting them using the source locations found with Load above

That way there's no special casing of the local module.